crates/core/src/lock.rs

```diff
@@ -300,7 +300,7 @@ pub fn generate_lockfile_contents(mut entries: Vec<LockEntry>) -> String {
 /// The entries are sorted by name before being written back to the file.
 pub fn add_to_lockfile(entry: LockEntry, path: impl AsRef<Path>) -> Result<()> {
     let mut lockfile = read_lockfile(&path)?;
-    if let Some(index) = lockfile.entries.iter().position(|e| e.name() == entry.name()) {
+    if let Some(index) = lockfile.entries.par_iter().position_any(|e| e.name() == entry.name()) {
```

Comment:
The parallel iterator is detrimental here

![image](https://github.com/user-attachments/assets/62a91e39-26b2-4ea5-babe-40a279924272)